adhocteam/Head-Start-TTADP
# Form navigation more consistent

**Description of change**
 * Any page navigation saves the form to the API
 * Continue is not blocked by an invalid form
 * If the page is valid on navigation that page is marked as complete
 * Any save with an id of "new" triggers a page reload with the saved activity report's id instead of "new"
 * After visiting the review page validation errors are shown in the form
 * Fixed a bug where report recipients couldn't be removed

**How to test**
1) Pull down changes
2) Create a new report. Click "Continue" without completely filling out the first page, you should navigate to the second page
3) Visit the review page, and back to the first page. You should see required fields marked with errors
4) Completely fill out the first page. Navigate away with the side nav. The state of the first page should be "Complete"

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/313

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
